sql/src/test/java/org/apache/druid/sql/calcite/schema/DruidSchemaTest.java

```diff
@@ -239,6 +246,62 @@ public void testGetTableMapFoo2()
     Assert.assertEquals(SqlTypeName.BIGINT, fields.get(2).getType().getSqlTypeName());
   }
 
+  /**
+   * This tests that {@link SegmentMetadataHolder#getNumRows()} is correct in case
+   * of multiple replicas i.e. when {@link DruidSchema#addSegment(DruidServerMetadata, DataSegment)}
+   * is called more than once for same segment
+   */
+  @Test
+  public void testSegmentMetadataHolderNumRows()
+  {
+    Map<DataSegment, SegmentMetadataHolder> segmentsMetadata = schema.getSegmentMetadata();
+    final Set<DataSegment> segments = segmentsMetadata.keySet();
+    Assert.assertEquals(3, segments.size());
+    // find the only segment with datasource "foo2"
+    final DataSegment existingSegment = segments.stream()
+                                                .filter(segment -> segment.getDataSource().equals("foo2"))
+                                                .findFirst()
+                                                .orElse(null);
+    Assert.assertNotNull(existingSegment);
+    final SegmentMetadataHolder existingHolder = segmentsMetadata.get(existingSegment);
+    //update SegmentMetadataHolder of existingSegment with numRows=5
```

Comment:
nit: please make the style of comments consistent. I think it's better to add a space after `//` because most of comments do.

Reply:
Same for other comments.